ledger-service/http-json/src/main/scala/com/digitalasset/http/WebSocketService.scala

```diff
@@ -853,13 +853,20 @@ class WebSocketService(
             jwtPayload.parties,
             offPrefix,
             rq.q: q,
-          )
+          ).watchTermination() { (m, fd) =>
+            fd.onComplete(_ => logger.info("S11 finished in FMM"))
+            m
+          }
```

Comment:
In the docs:

> Completes when upstream completes and all consumed substreams complete
> Cancels when downstream cancels

When would `getTransactionSourceForParty` complete? maybe what is necessary is to cancel the sources on completion of the `wsMessageHandler`.